### ISSUE TITLE
run_actor_critic_from_arrays returns policies[t] evaluated at next_observations[t] after the update, not the distribution that produced actions[t]

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#382]
## Summary

`run_actor_critic_from_arrays` (`alberta_framework/core/actor_critic.py`) documents `ActorCriticArrayResult.policies` as "Per-step policy probabilities" alongside `actions` ("Per-step actions") and `values` ("Per-step previous-state value estimates"). `actions[t]` is sampled at `observations[t]` and `values[t]` is the value at `observations[t]`, but the scan emits `result.policy` — `ActorCriticUpdateResult.policy`, documented as the policy "at the new observation" — i.e. the **post-update** softmax at `next_observations[t]`. The three per-step arrays are not on the same time index. The sibling continuous runner already records `current_mean, current_sigma` at `obs`, aligned with `actions[t]`.

Any per-step policy diagnostic computed from the returned arrays — `log π(a_t|s_t)`, policy entropy at the visited state, KL between arms, importance ratios for off-policy correction — reads the wrong row.

## Falsifier (deterministic, pristine `main` @ `4f75cc8`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.actor_critic import ActorCriticAgent, ActorCriticConfig, run_actor_critic_from_arrays
agent = ActorCriticAgent(ActorCriticConfig(n_actions=3, actor_step_size=0.5, gamma=0.9))
state = agent.init(feature_dim=2, key=jr.key(5)).replace(
    actor_weights=jnp.array([[3.0, 0.0], [0.0, 3.0], [0.0, 0.0]]))
obs = jnp.array([[1., 0.], [0., 1.], [1., 0.]]); nxt = jnp.array([[0., 1.], [1., 0.], [0., 1.]])
r = run_actor_critic_from_arrays(agent, state, obs, jnp.ones(3), jnp.array([False]*3), nxt)
print("actions:", r.actions)
print("policies[0]:", r.policies[0], " policy at observations[0]:", agent.policy(state, obs[0]))
```

Output on `main`:

```
actions: [0 1 0]
policies[0]: [0.0452785 0.909443  0.0452785]   policy at observations[0]: [0.909443  0.0452785 0.0452785]
```

`policies[0]` puts 0.045 on the action that was actually taken with probability 0.909; `log π(a_t|s_t)` from the returned arrays is `[-3.09, -0.72, -3.09]` against a ground truth of `[-0.09, -0.09, -0.72]` — wrong by 3 nats on two of three steps, with a sign that flips with the trajectory.

Expected: `policies[t]` is the pre-update policy at `observations[t]` (the distribution that produced `actions[t]`), as the continuous runner already does; on a rejected update it is neutralized like `actions[t]`.

## Dedup

`gh issue/pr list --search "run_actor_critic_from_arrays" / "policies"`: nothing on this surface (adjacent AC PRs #202, #64 are the 0×inf lane). No open PR touches `core/actor_critic.py`.

Fix PR follows (failing-test-first).
